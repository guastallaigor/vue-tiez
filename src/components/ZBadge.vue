<template>
  <span class="z-badge" :class="[color, {'with-text': withText}]">{{ content }}</span>
</template>

<script>
export default {
  name: 'z-badge',
  props: {
    content: {
      type: String,
      required: true,
      default: ''
    },
    withText: {
      type: Boolean,
      default: false
    },
    color: {
      type: String,
      default: 'black',
      validator: (v) => (['red', 'black', 'dark'].includes(v))
    }
  }
}
</script>

<style lang="scss">
.z-badge {
  color: rgb(255, 255, 255);
  display: inline-block;
  font-size: 10px;
  font-weight: bold;
  line-height: 15px;
  height: 15px;
  min-width: 15px;
  text-align: center;
  padding: 0px 4px;
  border-radius: 7px;

  &.with-text {
    font-weight: normal;
    text-transform: uppercase;
    min-height: 15px;
    padding: 4px 10px;
    border-radius: 20px;
  }


  &.black {
    background: black;
  }

  &.red {
    background: rgb(255, 0, 0);
  }

  &.dark {
    color: rgb(0, 0, 0);
    background: white;
  }
}
</style>
