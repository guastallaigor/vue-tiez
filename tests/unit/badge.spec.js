import { shallowMount } from '@vue/test-utils'
import ZBadge from '../../src/components/ZBadge'

describe('When I create the ZBadge component', () => {
  const createComponent = (propsData = {}, slot) => {
    return shallowMount(ZBadge, {
      propsData,
      slots: {
        default: slot || 'ZBadge'
      }
    })
  }

  it('should be a Vue instance', () => {
    const wrapper = createComponent({ content: 10 })
    expect(wrapper.find(ZBadge).isVueInstance()).toBe(true)
  })

  it('should have a number content and default color', () => {
    const wrapper = createComponent({ content: 10 })
    const badge = wrapper.find('.z-badge')
    expect(badge.exists()).toBe(true)
    expect(badge.classes().length).toBe(2)
    expect(badge.classes()).toEqual(['z-badge', 'black'])
    expect(badge.html()).toBe('<span class="z-badge black">10</span>')
  })

  it('should have a number content and a black color', () => {
    const wrapper = createComponent({ content: 10, color: 'black' })
    const badge = wrapper.find('.z-badge')
    expect(badge.exists()).toBe(true)
    expect(badge.classes().length).toBe(2)
    expect(badge.classes()).toEqual(['z-badge', 'black'])
    expect(badge.html()).toBe('<span class="z-badge black">10</span>')
  })

  it('should have a number content and a red color', () => {
    const wrapper = createComponent({ content: 10, color: 'red' })
    const badge = wrapper.find('.z-badge')
    expect(badge.exists()).toBe(true)
    expect(badge.classes().length).toBe(2)
    expect(badge.classes()).toEqual(['z-badge', 'red'])
    expect(badge.html()).toBe('<span class="z-badge red">10</span>')
  })

  it('should have a number content and a dark color', () => {
    const wrapper = createComponent({ content: 10, color: 'dark' })
    const badge = wrapper.find('.z-badge')
    expect(badge.exists()).toBe(true)
    expect(badge.classes().length).toBe(2)
    expect(badge.classes()).toEqual(['z-badge', 'dark'])
    expect(badge.html()).toBe('<span class="z-badge dark">10</span>')
  })

  it('should have a text content and a withText prop passed', () => {
    const wrapper = createComponent({ content: 'Content', withText: true })
    const badge = wrapper.find('.z-badge')
    expect(badge.exists()).toBe(true)
    expect(badge.classes().length).toBe(3)
    expect(badge.classes()).toEqual(['z-badge', 'black', 'with-text'])
    expect(badge.html()).toBe('<span class="z-badge black with-text">Content</span>')
  })

  it('should match snapshot', () => {
    const wrapper = createComponent({
      content: 'Content',
      withText: true,
      color: 'black'
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
