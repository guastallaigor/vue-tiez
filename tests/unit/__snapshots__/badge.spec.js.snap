// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I create the ZBadge component should match snapshot 1`] = `<span class="z-badge black with-text">Content</span>`;
